dcc-storage-client/src/main/java/org/icgc/dcc/storage/client/transport/MemoryMappedParallelPartObjectTransport.java

```diff
@@ -115,13 +115,11 @@ public Part call() throws Exception {
                   log.debug("Fail checksum. Reupload part: {}", part);
                   progress.startTransfer();
                   proxy.uploadPart(channel, part, objectId, uploadId);
-                  progress.incrementBytesWritten(part.getPartSize());
                 }
                 progress.incrementChecksumParts();
               } else {
                 progress.startTransfer();
                 proxy.uploadPart(channel, part, objectId, uploadId);
-                progress.incrementBytesWritten(part.getPartSize());
```

Comment:
Ok, keep in mind that the ProgressInputStream is shared between implementations of transport. So you have to ensure that the other implementations continue to work correctly.